crates/zk-por-core/src/e2e.rs

```diff
@@ -48,61 +102,46 @@ pub fn recursive_prove_subproofs<const RECURSION_BRANCHOUT_NUM: usize>(
     let mut last_level_proofs = subproofs;
     let recursive_levels = circuit_registry.get_recursive_levels();
 
-    for level in 0..recursive_levels {
+    for level in 1..=recursive_levels {
         let start = std::time::Instant::now();
+
         let last_level_vd_digest = last_level_circuit_vd.circuit_digest;
         let last_level_empty_proof =
             circuit_registry.get_empty_proof(&last_level_vd_digest).expect(
```

Comment:
[suggestion] can clone outside `get_empty_proof`; the meaning of get is for obtaining a reference